clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -2775,7 +2777,7 @@ raw2trace_t::insert_post_chunk_encodings(raw2trace_thread_data_t *tdata,
 // extra processing here).
 std::string
 raw2trace_t::write(raw2trace_thread_data_t *tdata, const trace_entry_t *start,
-                   const trace_entry_t *end, std::vector<app_pc> decode_pcs)
+                   const trace_entry_t *end, app_pc *decode_pcs, size_t decode_pcs_size)
```

Comment:
If this is about the vector copy could we just pass the vector by reference?

Reply:
There are some cases when vector is null and we could not use reference in this case.
PR #6176